beacon-chain/blockchain/process_block.go

```diff
@@ -185,13 +208,8 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 			}
 		}()
 	}
-	justified := s.cfg.ForkChoiceStore.JustifiedCheckpoint()
-	start := time.Now()
-	headRoot, err := s.cfg.ForkChoiceStore.Head(ctx)
-	if err != nil {
-		log.WithError(err).Warn("Could not update head")
-	}
 	if blockRoot != headRoot {
```

Comment:
I think you have to fetch head from forkchoice here. You moved `forkchoiceUpdateWithExecution` above, meaning that the head can change (on a side note, this looks like a possibly unwanted change in logic). Actually I see that you ignore the boolean return value from `forkchoiceUpdateWithExecution` - you can use it to determine whether the head has changed.